packages/contracts/src/contracts/current/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -76,77 +76,148 @@ contract MixinExchangeCore is
     /*
     * Core exchange functions
     */
-
-    /// @dev Fills the input order.
-    /// @param order Order struct containing order specifications.
-    /// @param takerTokenFillAmount Desired amount of takerToken to fill.
-    /// @param signature Proof of signing order by maker.
-    /// @return Total amount of takerToken filled in trade.
-    function fillOrder(
-        Order order,
-        uint256 takerTokenFillAmount,
+    
+    function orderStatus(
+        Order memory order,
         bytes signature)
         public
```

Comment:
We should check gas differences of `public` vs `internal` here. AFAIK, these in memory arguments will only be passed by reference for internal functions.